boa/src/vm/mod.rs

```diff
@@ -103,11 +101,14 @@ impl Context {
             }};
         }
 
-        let opcode = self.vm.frame().code.code[self.vm.frame().pc]
+        let opcode: Opcode = self.vm.frame().code.code[self.vm.frame().pc]
             .try_into()
             .unwrap();
         self.vm.frame_mut().pc += 1;
```

Comment:
This part is not being profiled. Could we somehow add a "opcode retrieval" profiling section?

Also, this `unwrap()` should be changed to a `expect`, with some message, and I would like some explanation on why this can't fail. Also, using indexing here can panic, and has bound checks which we might be able to avoid somehow, not sure how, though. In the worst case, I would use `get()` and `expect()`.

Reply:
I did some of the changes, I will open some issues for the rest.